homeassistant/components/google_photos/__init__.py

```diff
@@ -41,6 +54,11 @@ async def async_setup_entry(
     except ClientError as err:
         raise ConfigEntryNotReady from err
     entry.runtime_data = auth
+
+    scopes = entry.data["token"]["scope"].split(" ")
+    if any(scope == UPLOAD_SCOPE for scope in scopes):
+        async_register_services(hass)
```

Comment:
I'd rather always register the devices in async setup and then raise a proper error message if it's not possible

Reply:
Done, moved the check to be during the service call and added a new error message and test.